### PR TITLE
[PROP-869] implemented assertion and tests

### PR DESCRIPTION
**Pull request by harispericorn**

### CODE REVIEW

File: tests/propbase_staking_tests.move [hunk 1]
```diff
@@ -2998,6 +3040,48 @@ module propbase::propbase_staking_tests {
...
+        vector::push_back(&mut receivers, signer::address_of(address_2));
+        setup_prop(resource, receivers);
+
+        let required_funds = (((250000 - 80000) / 100)* (20000000000/31622400)) * 50 ;
+        propbase_staking::set_treasury(admin, signer::address_of(address_2));
+        propbase_staking::set_reward_treasurer(admin, signer::address_of(address_1));
+        propbase_staking::add_reward_funds<PROPS>(address_1, required_funds);
+
+        propbase_staking::create_or_update_stake_pool(admin,string::utf8(b"Hello"), 20000000000, 80000, 250000, 15, 50, 1000000000, 10000000000, 31622400, update_config);
+        fast_forward_secs(10000);
```
Comment (neethu-satheesh-s) [inline tests/propbase_staking_tests.move:3080]: @harispericorn  Hi Hari, looks great.
Lets also add a test to check on the edge case of exact pool start time.

**Review by neethu-satheesh-s — CHANGES_REQUESTED** [review]

File: tests/propbase_staking_tests.move (hunk 1 above)
Comment (harispericorn) [reply tests/propbase_staking_tests.move:3080]: ok added

**Review by neethu-satheesh-s — APPROVED** [review]
